crates/proto/src/rr/rdata/cert.rs

```diff
@@ -0,0 +1,161 @@
+// Copyright 2024 Brian Taber <btaber@zsd.systems>
+//
+// Licensed under the Apache License, Version 2.0, <LICENSE-APACHE or
+// https://apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT or
+// https://opensource.org/licenses/MIT>, at your option. This file may not be
+// copied, modified, or distributed except according to those terms.
+
+//! CERT record type for storing certificates in DNS
+use std::fmt;
+
+#[cfg(feature = "serde")]
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    error::*,
+    rr::{RData, RecordData, RecordDataDecodable, RecordType},
+    serialize::binary::*,
+};
+
+/// [RFC 4398, Storing Certificates in DNS, November 1987][rfc4398]
+/// https://tools.ietf.org/html/rfc4398
+///
+/// ```text
+///
+/// The CERT resource record (RR) has the structure given below.  Its RR
+/// type code is 37.
+///
+///    1 1 1 1 1 1 1 1 1 1 2 2 2 2 2 2 2 2 2 2 3 3
+/// 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+/// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+/// |             type              |             key tag           |
+/// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+/// |   algorithm   |                                               /
+/// +---------------+            certificate or CRL                 /
+/// /                                                               /
+/// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-|
+/// ```
+#[cfg_attr(feature = "serde", derive(Deserialize, Serialize))]
+#[derive(Default, Debug, PartialEq, Eq, Hash, Clone)]
+pub struct CERT {
+    cert_type: u32,
+    key_tag: u32,
+    algorithm: u32,
```

Comment:
`cert_type`, `key_tag` and `algorithm` look like the kinds of things we'd normally want to use `enum`s for, why not do so here?

Reply:
Given the context here, I'd recommend:
* use u16 for keytag
* for algorithm, use the existing Algorithm enum from crates/proto/src/rr/dnssec/algorithm.rs (it has an unknown algorithm variant and from_u8 on that enum will treat '0' as 'unknown'.)
* Make a new enum for cert_type, appropriate to this RR.

Reply:
I was also unsure if not having dnssec feature would cause issues, so implemented algorithm in cert mod, should I go remove that and use from dnssec mod?